src/main/kotlin/ar/edu/unahur/obj2/caralibro/Usuario.kt

```diff
@@ -1,11 +1,35 @@
 package ar.edu.unahur.obj2.caralibro
 
+import java.lang.Exception
+
 class Usuario {
   val publicaciones = mutableListOf<Publicacion>()
+  val amigos = mutableListOf<Usuario>()
+
 
   fun agregarPublicacion(publicacion: Publicacion) {
     publicaciones.add(publicacion)
   }
 
+  fun agregarAmigo(amigo: Usuario) {amigos.add(amigo)}
+
+  fun cantidadAmigos() = amigos.size
+
   fun espacioDePublicaciones() = publicaciones.sumBy { it.espacioQueOcupa() }
+
+  fun esMasAmistoso_Que_(unUsuario: Usuario, otroUsuario: Usuario) {
+    unUsuario.amigos.size > otroUsuario.amigos.size
+  }
+
+  fun mejoresAmigos() = amigos.filter { a -> this.puedenVerTodasLasPublicaciones(a) }
+
+  fun puedenVerTodasLasPublicaciones(usuario: Usuario) = publicaciones.all { p -> p.puedeVerPublicacion(usuario) }
```

Comment:
Sería mejor nombrarlo en singular, porque se refiere a **un** usuario:

```suggestion
  fun puedeVerTodasLasPublicaciones(usuario: Usuario) = publicaciones.all { p -> p.puedeVerPublicacion(usuario) }
```